File: ZakaZaka/ClientApp/src/app/Model/restaurant.ts
import {RestaurantCuisines} from "./restaurantCuisine";
import {RestaurantFood} from "./restaurantFood";
import {RestaurantReview} from "./restaurantReview";

export class Restaurant{
  constructor(
    public id?: number,
    public name?: string,
    public pathToImage?: string,
    public minimumOrder?: number,
    public costDelivery?: number,
    public timeToDelivery?: number,
    public payToCard?: boolean,
    public restaurantCuisines?: RestaurantCuisines[],
    public restaurantFoods?: RestaurantFood[],
    public restaurantReviews?: RestaurantReview[]
  ) {}
}

File: ZakaZaka/Models/Restaurants/Restaurant.cs
﻿using System.Collections.Generic;

namespace ZakaZaka.Models.Restaurants
{
    public class Restaurant
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string PathToImage { get; set; }
        public int MinimumOrder { get; set; }
        public int CostDelivery { get; set; }
        public double TimeToDelivery { get; set; }
        public bool PayToCard { get; set; }
        public List<RestaurantCuisine> RestaurantCuisines { get; set; } = new List<RestaurantCuisine>();
        public List<RestaurantFood> RestaurantFoods { get; set; } = new List<RestaurantFood>();
        public List<RestaurantReview> RestaurantReviews { get; set; } = new List<RestaurantReview>();
    }
}